src/test/java/ezvcard/io/json/roundtrip/JCardRoundTripTest.java

```diff
@@ -0,0 +1,155 @@
+			return name.toLowerCase().endsWith(ext);
+		}
+	}
+
+	@Rule
+	public TemporaryFolder tempFolder = new TemporaryFolder();
+
+	@Test
+	public void convert_to_jcard() throws Throwable {
+		File input = new File(JCardRoundTripTest.class.getResource("").toURI());
+		File[] dir = input.listFiles(new ExtensionFilter(".vcf"));
+		for (File vcf : dir) {
+			String file = vcf.getName().toString();
+			if (file.startsWith("John_Doe_EVOLUTION") || file.startsWith("John_Doe_IPHONE")
+					|| file.startsWith("John_Doe_MAC_ADDRESS_BOOK") || file.toLowerCase().contains("outlook")) {
+				continue;
+			}
+			assertEquals(file, read(file.replace(".vcf", ".json")), convertVCard(vcf));
+		}
+	}
+
+	@Test
+	public void convert_from_jcard() throws Throwable {
+		File input = new File(JCardRoundTripTest.class.getResource("").toURI());
+		File[] dir = input.listFiles(new ExtensionFilter(".json"));
+		for (File jcard : dir) {
+			String file = jcard.getName().toString();
+			assertEquals(file, read(file.replace(".json", ".vcf")), convertJCard(jcard));
+		}
+	}
+
+	/**
+	 * Run this if there are new vcard samples added to
+	 * src/test/resources/ezvcard/io/text/, or if the output format changes.
+	 * 
+	 * @throws Throwable
+	 */
+	public void update_samples() throws Throwable {
+		File input = new File(SampleVCardsTest.class.getResource("").toURI());
+		File[] dir = input.listFiles(new ExtensionFilter(".vcf"));
+		for (File vcf : dir) {
+			String file = vcf.getName().toString();
+			write(file.replace(".vcf", ".json"), convertVCard(vcf));
+		}
+
+		input = new File(JCardRoundTripTest.class.getResource("").toURI());
+		dir = input.listFiles(new ExtensionFilter(".json"));
+		for (File jcard : dir) {
+			String file = jcard.getName().toString();
+			write(file.replace(".json", ".vcf"), convertJCard(jcard));
+		}
+	}
+
+	private static String convertJCard(File jcard) throws IOException {
+		StringWriter sw = new StringWriter();
+		VCardWriter writer = new VCardWriter(sw, VCardVersion.V4_0);
+
+		try {
+			for (VCard vcard : new JCardReader(jcard).readAll()) {
+				writer.write(vcard);
+			}
+		} finally {
+			writer.close();
+		}
+		return sw.toString();
+	}
+
+	public static String convertVCard(File vcf) throws IOException, FileNotFoundException {
+		StringWriter sw = new StringWriter();
+		JCardWriter writer = new JCardWriter(sw);
+		writer.setIndent(true);
+
+		try {
+			for (VCard vcard : new VCardReader(vcf).readAll()) {
+				writer.write(vcard);
+			}
+		} finally {
+			writer.close();
+		}
+		return sw.toString();
+	}
+
+	private static String read(String fileName) throws IOException {
+		return IOUtils.toString(IOUtils.utf8Reader(JCardRoundTripTest.class.getResourceAsStream(fileName)));
+	}
+
+	private static void write(String fileName, String converted) throws IOException {
+		File file = new File("src/test/resources/ezvcard/io/json/roundtrip/" + fileName);
+		Writer writer = new FileWriter(file);
+		try {
+			writer.write(converted);
+			writer.flush();
```

Comment:
The call to `flush()` is not necessary.  The writer will be flushed when it is closed.